packages/flutter/lib/src/material/checkbox_theme.dart

```diff
@@ -124,6 +136,7 @@ class CheckboxThemeData with Diagnosticable {
       splashRadius: lerpDouble(a?.splashRadius, b?.splashRadius, t),
       materialTapTargetSize: t < 0.5 ? a?.materialTapTargetSize : b?.materialTapTargetSize,
       visualDensity: t < 0.5 ? a?.visualDensity : b?.visualDensity,
+      shape: ShapeBorder.lerp(a?.shape, b?.shape, t) as RoundedRectangleBorder?,
```

Comment:
If we're going to support textDirection as a parameter, then we might as well lerp it here.